### DISCUSSION TITLE
osm2pgsql-replication init - putting DB password into command

### DISCUSSION THREAD

**Discussion by serpikovs** [osm2pgsql-dev/osm2pgsql#1747] [Q&A] [answered]
Hello! 

Try to do this:
``` 
#!/bin/bash

HOST='localhost';
PORT='5435'; 
DB='osm'; 
USER='postgres'; 
PASSWORD='qwertyasdf123'

osm2pgsql-replication init -d $DB -H $HOST -P $PORT -U $USER
```

and got next err:
```
root@ubuntu:/opt/docker/osm2pgsql# ./update.sh 
Traceback (most recent call last):
  File "/usr/local/bin/osm2pgsql-replication", line 535, in <module>
    sys.exit(main())
  File "/usr/local/bin/osm2pgsql-replication", line 527, in main
    conn = connect(args)
  File "/usr/local/bin/osm2pgsql-replication", line 82, in connect
    return psycopg.connect(dbname=args.database, user=args.username,
  File "/usr/lib/python3/dist-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "localhost" (127.0.0.1), port 5435 failed: fe_sendauth: no password supplied
```

In [docs](https://osm2pgsql.org/doc/man/osm2pgsql-replication-1.6.0.html#OPTIONS%20%E2%80%99osm2pgsql-replication%20init%E2%80%99) I cannot find password parameter, also there is no transfer here:
[https://github.com/openstreetmap/osm2pgsql/blob/08d1e1cdcd075afff9e56894ae26ffe9a6bc303b/scripts/osm2pgsql-replication#L82](https://github.com/openstreetmap/osm2pgsql/blob/08d1e1cdcd075afff9e56894ae26ffe9a6bc303b/scripts/osm2pgsql-replication#L82)

Additionally, --password (or -W) argument of ```osm2pgsql-replication init``` doesnt work

Could you highlight solution?

**Comment by joto** [comment, marked as answer]
The option `-W` doesn't take the password as argument on the command line, it means "ask for password interactively". Please read https://osm2pgsql.org/doc/manual.html#database-connection for details on all the different ways of specifying how to connect to the database.

Putting passwords in scripts isn't something I'd recommend.

**Reply by serpikovs** [reply]
@joto , thanks! The connection string works for me!

``` postgresql://[user[:password]@][netloc][:port][,...][/dbname][?param1=value1&...]```

```osm2pgsql-replication init -d 'postgresql://postgres:qwertyasdf123@localhost:5435/osm'```

**Reply by lonvia** [reply]
Password in the command line is a bad idea. The recommended way by PostgreSQL is a [pgpass file](https://www.postgresql.org/docs/current/libpq-pgpass.html).
